src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.HillClimbing.cs

```diff
@@ -378,9 +394,28 @@ private void ChangeThreadCount(int newThreadCount, StateOrTransition state)
                 _completionsSinceLastChange = 0;
             }
 
-            private void LogTransition(int newThreadCount, double throughput, StateOrTransition state)
+            private void LogTransition(int newThreadCount, double throughput, StateOrTransition stateOrTransition)
             {
-                // TODO: Log transitions
+                // Use the _log array as a circular array for log entries
+                int index = (_logStart + _logSize) % LogCapacity;
+
+                if(_logSize == LogCapacity)
+                {
+                    _logStart = (_logStart + 1) % LogCapacity;
+                    _logSize--; // hide this slot while we update it
```

Comment:
This comment suggests that this is supposed to be thread safe. How does it work?